Linking problems using default Ubuntu 18.04 `gcc-arm-none-eabi` toolchain

There's an error with `arm-none-eabi-ld` when trying to build the mbed firmware:
```
igvc-firmware/cmake-build-debug/src/mbed/igvc-firmware-mbed_pp.link_script.ld:80 cannot move location counter backwards (from 000000001000f738 to 0000000010007be0)
collect2: error: ld returned 1 exit status
```

--- a/README.md
+++ b/README.md
@@ -7,6 +7,11 @@ Welcome to the RoboJackets IGVC firmware repo!
 * protobuf-compiler
 * python-protobuf
 
+The second two can be installed with
+```bash
+sudo apt install -y protobuf-compiler python-protobuf
+```
+
 For more details, check the [Dockerfile](Dockerfile).
 
 ### Updating `arm-none-eabi-gcc`

--- a/README.md
+++ b/README.md
@@ -3,10 +3,25 @@
 Welcome to the RoboJackets IGVC firmware repo!
 
 ## Dependencies
+* Updated `arm-none-eabi-gcc`
 * protobuf-compiler
 * python-protobuf
 
 For more details, check the [Dockerfile](Dockerfile).
+
+### Updating `arm-none-eabi-gcc`
+For some reason, it seems like the version of `arm-none-eabi-ld` on Ubuntu 18.04 has some linking problems.
+The problem seems to go away on newer versions of `arm-non-eabi-gcc` though.
+
+**Steps to update `arm-none-eabi-gcc`**:
+1. Download [gcc-arm-none-eabi-8-2018-q4-major-linux.tar.bz2](https://developer.arm.com/-/media/Files/downloads/gnu-rm/8-2018q4/gcc-arm-none-eabi-8-2018-q4-major-linux.tar.bz2?revision=d830f9dd-cd4f-406d-8672-cca9210dd220?product=GNU%20Arm%20Embedded%20Toolchain,64-bit,,Linux,8-2018-q4-major)
+2. Untar it and move it to some location (eg. `/opt/gcc-arm-none-eabi-8-2018-q4-major`)
+3. Add it to your `$PATH` variable, eg. add the following line to `.bashrc`:
+   ```bash
+   export PATH="/opt/gcc-arm-none-eabi-8-2018-q4-major/bin:$PATH"
+   ```
+
+This should solve the `ld` issues
 
 ## Building Documentation
 Make sure to `git submodule init && git submodule update` as the repo uses git submodules for the `mbed-os` and `nanopb` dependencies.

